clayui.com/src/components/IconSearch/index.js

```diff
@@ -6,74 +6,77 @@
 
 import ClayForm, {ClayInput} from '@clayui/form';
 import ClayIcon from '@clayui/icon';
-import React, {useEffect, useState} from 'react';
+import React, {useMemo, useState} from 'react';
 
 const spritemap = '/images/icons/icons.svg';
 
-const IconSearch = source => {
+/**
+ * Component that allows searching through icons
+ * @param {string} label - Label of the search input field.
+ * @param {string} placeholder - Placeholder of the search input field.
+ * @param {Array[Object]} source - Path to the source JSON. Needs to follow pattern of [{name: String, aliases: Array<String>}]
+ * @param {string} type - Type of icons being provided, supports "icons" and "flags"
+ */
+const IconSearch = ({
+	label = 'Search Icons',
+	placeholder = 'Search Icons...',
+	source,
+	type = 'icons',
+}) => {
 	const [searchQuery, setSearchQuery] = useState('');
-	const [filteredIconNames, setFilteredIconNames] = useState([]);
-	const list = filteredIconNames.length ? filteredIconNames : [];
-
-	source = [...source.source];
-
-	useEffect(() => {
-		const filteredIcons = source.filter(icon => {
-			const iconPrimary = icon[Object.keys(icon)[0]].toUpperCase();
-			let iconSecondary = icon[Object.keys(icon)[1]];
-
-			iconSecondary = Array.isArray(iconSecondary)
-				? iconSecondary.map(value => value.toUpperCase())
-				: iconSecondary.toUpperCase();
+	let list = [];
 
-			return (
-				iconPrimary.includes(searchQuery.toUpperCase()) ||
-				iconSecondary.includes(searchQuery.toUpperCase())
-			);
-		});
+	const filteredIcons = useMemo(() => {
+		const query = searchQuery.toLowerCase();
 
-		setFilteredIconNames(filteredIcons);
+		return source.filter(
+			({aliases, name}) =>
+				aliases.some(alias => alias.toLowerCase().includes(query)) ||
+				name.toLowerCase().includes(query)
```

Comment:
One little nit here @kresimir-coko: when you have a chain of conditional expressions like this, you generally want to put the cheapest one first, to maximize the chances of short-circuiting and bailing out of the computation early. In this example, the `name` check is cheaper (two function calls) than the `aliases` one (one function call plus two more for each alias in the list), which is why I put it first in my suggestion to you.

This kind of thing is microoptimization .... until it isn't. 😂 So even though it doesn't matter in this case, it's a good habit to get into.

Reply:
Ah, I was going for alphabetical sorting, I'll force-push with them organized in the manner you proposed! Thanks for the heads up